src/services/apimPolicyBuilder.ts

```diff
@@ -168,6 +209,35 @@ export class ApimPolicyBuilder {
     }
   }
 
+  /**
+   * Updates the matching policy or appends a new policy
+   * @param policyType The policy type (inbound | outbound | backend | on-error)
+   * @param policyName The xml element name of the policy
+   * @param newPolicy The new policy XML object
+   * @param predicate The condition to check for existing policy
+   */
+  private updatePolicy(policyType: PolicyType, policyName: string, newPolicy: any, predicate: (policy: any) => boolean): void {
+    const existing = this.policyRoot.policies[policyType][0][policyName];
+
+    // Add new policy to empty list
+    if (!existing) {
+      this.policyRoot.policies[policyType][0][policyName] = [newPolicy];
```

Comment:
Nit: wonder if there's anyway to simplify this?

Reply:
Which part(s) seem complex?  This function basically handles either updating an existing policy or appending a new policy.

Reply:
it's not complex per-se, just not a fan of 2+ level deep access :) `this.policyRoot.policies[policyType][0][policyName]`

Reply:
Understood - but the XML schema for the policies is out of our control here.  Going to leave as is for now.